python/pyarrow/tests/strategies.py

```diff
@@ -261,17 +274,23 @@ def arrays(draw, type, size=None, nullable=True):
     elif pa.types.is_date(ty):
         value = st.dates()
     elif pa.types.is_timestamp(ty):
+        if zoneinfo is None:
+            pytest.skip('no module named zoneinfo')
+        if ty.tz is None:
+            pytest.skip('requires timezone not None')
```

Comment:
I think we can also create a datetime without timezone in this case, instead of skipping the whole test? (that might require rearranging the code below a bit)

Reply:
Although I now see some previous comments about that below ...

Reply:
We can maybe try to tackle this as a follow-up JIRA?